enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -721,14 +721,18 @@ def can_redeem(self, lms_user_id, content_key, skip_customer_user_check=False):
         return (True, None, existing_redemptions)
 
     def credit_available(self, lms_user_id=None):
-        if self.remaining_balance_per_user(lms_user_id) > 0:
-            return True
-        return False
+        remaining_balance_per_user = self.remaining_balance_per_user(lms_user_id) or 0
```

Comment:
Similar to handling `None` versus `0` differently in the `remaining_balance_per_user` methods, the `None` or `0` should likely be handled as semantically different here as well. If `self.remaining_balance_per_user` returns `None` because there is no `self.per_learner_enrollment_limit` set, I don't believe it's semantically correct to fallback to say the user has a remaining balance of `0`, as there is no per-learner limit (this method would return `False` currently, even when there is no limit set).

Related, are there additional checks that are relevant to determine whether these policies are "available" beyond only checking against the per-learner limits? Would cases such as the following be handled by these `credit_available` methods?

* User hasn't exceeded their individual per-learner limit, but the policy has already exceeded its configured `spend_limit` (not redeemable).
* User hasn't exceeded their individual per-learner limit, but the associated subsidy has no remaining balance.
* User hasn't exceeded their individual per-learner limit, but the policy is no longer active.
* User hasn't exceeded their individual per-learner limit, but the associated subsidy is expired.

These are some of the additional checks that the `can_redeem` methods perform that might be worth referring to. Just verifying that the `credits_available` API is taking into account all factors that could deem a policy available or unavailable.

Reply:
> Similar to handling None versus 0 differently in the remaining_balance_per_user methods, the None or 0 should likely be handled as semantically different here as well.
> 
> If self.remaining_balance_per_user returns None because there is no self.per_learner_enrollment_limit set, I don't believe it's semantically correct to fallback to say the user has a remaining balance of 0, as there is no per-learner limit (this method would return False currently, even when there is no limit set).
> 

@adamstankiewicz Agreed. I have handled that part in the new changes. 

> Related, are there additional checks that are relevant to determine whether these policies are "available" beyond only checking against the per-learner limits? Would cases such as the following be handled by these credit_available methods?
> 
> User hasn't exceeded their individual per-learner limit, but the policy has already exceeded its configured spend_limit (not redeemable).
> User hasn't exceeded their individual per-learner limit, but the associated subsidy has no remaining balance.
> User hasn't exceeded their individual per-learner limit, but the policy is no longer active.
> User hasn't exceeded their individual per-learner limit, but the associated subsidy is expired.
> These are some of the additional checks that the can_redeem methods perform that might be worth referring to. Just verifying that the credits_available API is taking into account all factors that could deem a policy available or unavailable.
> 

@adamstankiewicz I believe the existing code doesn't check any of these conditions. Would it be okay if I create a follow up ticket/PR to address this feedback as all of these checks would require adding unit tests..

Reply:
> @adamstankiewicz I believe the existing code doesn't check any of these conditions. Would it be okay if I create a follow up ticket/PR to address this feedback as all of these checks would require adding unit tests..

Yep, totally. Just wanted to call out that the current logic may not be fully taking into account all factors.